pkgs/servers/nosql/keydb/default.nix

```
@@ -0,0 +1,48 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, libuuid, curl
+, openssl
+, pkg-config
+, systemd
+}:
+
+stdenv.mkDerivation rec {
+  pname = "keydb";
+  version = "6.3.1";
+
+  src = fetchFromGitHub {
+    owner = "snapchat";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-00Dx6GRAXVpJUXhu4kjsVCEsoN615vhGvc7+gViERqI=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+  buildInputs = [
+    (curl.override { inherit openssl; })
+    libuuid
+    systemd
+    openssl
+  ];
+
+  makeFlags = [
+    "PREFIX=${placeholder "out"}"
+    "USE_SYSTEMD=yes"
+    "BUILD_TLS=yes"
+  ];
```

Fix cross-compilation:
```suggestion
    "AR:=$(AR)"
    "RANLIB:=$(RANLIB)"
    "USEASM=${if stdenv.isx86_64 then "true" else "false"}"
  ];
```